Parsers/ASimFileEvent/Parsers/vimFileEventSentinelOne.yaml

```diff
@@ -0,0 +1,118 @@
+Parser:
+  Title: File Event Parser for SentinelOne
+  Version: '0.1.0'
+  LastUpdated: July 24, 2023
+Product:
+  Name: SentinelOne
+Normalization:
+  Schema: FileEvent
+  Version: '0.2.1'
+References:
+- Title: ASIM File Event Schema
+  Link: https://aka.ms/ASimFileEventDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+- Title: SentinelOne Documentation
+- Link: https://<SOneInstanceDomain>.sentinelone.net/api-doc/overview
+Description: |
+  This ASIM parser supports normalizing SentinelOne logs to the ASIM File Event normalized schema. SentinelOne events are captured through SentinelOne data connector which ingests SentinelOne server objects such as Threats, Agents, Applications, Activities, Policies, Groups, and more events into Microsoft Sentinel through the REST API.
+ParserName: vimFileEventSentinlOne
+EquivalentBuiltInParser: _Im_FileEvent_SentinelOne
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let GetWindowsFilenamePart = (path: string) { tostring(split(path, @'\')[-1]) };
+  let GetLinuxFilenamePart = (path: string) { tostring(split(path, @'/')[-1]) };
+  let EventTypeLookup = datatable (alertInfo_eventType_s: string, EventType: string)
+      [
+      "FILECREATION", "FileCreated",
+      "FILEMODIFICATION", "FileModified",
+      "FILEDELETION", "FileDeleted",
+      "FILERENAME", "FileRenamed"
+  ];
+  let parser = (disabled: bool=false) {
+      let allFileData = SentinelOne_CL
+          | where not(disabled)
+              and event_name_s == "Alerts."
+              and alertInfo_eventType_s in ('FILECREATION', 'FILEMODIFICATION', 'FILEDELETION', 'FILERENAME');
+      let windowsFileData = allFileData
+          | where agentDetectionInfo_osFamily_s == "windows"
+          | extend
+              TargetFilePathType = "Windows Local",
+              TargetFileName = GetWindowsFilenamePart(targetProcessInfo_tgtFilePath_s),
+              SrcFileName = GetWindowsFilenamePart(targetProcessInfo_tgtFileOldPath_s);
+      let otherFileData = allFileData
+          | where agentDetectionInfo_osFamily_s != "windows"
+          | extend
+              TargetFilePathType = "Unix",
+              TargetFileName = GetLinuxFilenamePart(targetProcessInfo_tgtFilePath_s),
+              SrcFileName = GetLinuxFilenamePart(targetProcessInfo_tgtFileOldPath_s);
+      union windowsFileData, otherFileData
+      | lookup EventTypeLookup on alertInfo_eventType_s
+      | extend
+          EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s),
+          EventVendor = "SentinelOne",
+          EventProduct = "SentinelOne",
+          EventResult = "Success",
+          EventSchema = "FileEvent",
+          EventSchemaVersion = "0.2.1",
+          EventCount = toint(1),
+          DvcAction = "Allowed",
+          ActorUsername = sourceProcessInfo_user_s
```

Comment:
I noticed that ActorUsername is always having system name instead of user?
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/788788c2-04d2-4a0c-bf83-ac390a0093fe)

Reply:
can you check for user details too?

Reply:
like to know who is creating, modifying or deleting files

Reply:
The above data is generated is the live data of our organization and user related information is provided in above fashion for each user in org that's why we getting as above.